testing/smoke/lib.sh

```diff
@@ -227,14 +225,74 @@ legacy_ingest_pipelines() {
 }
 
 legacy_assertions() {
-    VERSION=${1}
-    ENTRIES=${2}
+    local VERSION=${1}
+    local ENTRIES=${2}
+    local LEGACY_ILM_POLICY=apm-rollover-30-days
     legacy_ingest_pipelines
     legacy_assert_templates ${VERSION}
     legacy_assert_ilm
     legacy_assert_events ${VERSION} ${ENTRIES}
 }
 
+data_stream_assertions() {
+    local VERSION=${1}
+    local ENTRIES=${2}
+    data_stream_assert_templates_ilm ${VERSION}
+    data_stream_assert_pipelines
+    data_stream_assert_events ${VERSION} ${ENTRIES}
+}
+
+data_stream_assert_pipelines() {
+    # NOTE(marclop) we could assert that the pipelines have some sort of version suffix
+    # in their name, however, the APM package version may not equal the deployment version.
+    echo "-> Asserting ingest pipelines..."
+    local SUCCESS=true
+    local RESPONSE=$(elasticsearch_curl '/_ingest/pipeline/*apm*')
+    local HAS_APM_PIPELINES=$(echo ${RESPONSE} | jq -r '.|length>0')
+    if [[ ${HAS_APM_PIPELINES} != true ]]; then
+        SUCCESS=false
+        echo "-> Did not find any APM ingest pipelines"
+        echo ${RESPONSE}
+    fi
+
+    if [[ ${SUCCESS} == false ]]; then
```

Comment:
IIUC, we can merge this condition block with `if [[ ${HAS_APM_PIPELINES} != true ]]; then` since `SUCCESS = false` will only happen if the first `if` evaluates to true